shell/platform/darwin/ios/framework/Headers/FlutterViewController.h

```diff
@@ -10,78 +10,123 @@
 
 #include "FlutterBinaryMessenger.h"
 #include "FlutterDartProject.h"
+#include "FlutterEngine.h"
 #include "FlutterMacros.h"
 #include "FlutterPlugin.h"
 #include "FlutterTexture.h"
 
+@class FlutterEngine;
+
+/**
+ * A `UIViewController` implementation for Flutter views.
+ *
+ * Dart execution, channel communication, texture registration, and plugin registration
+ * are all handled by `FlutterEngine`.  Calls on this class to those members all proxy
+ * through to the `FlutterEngine` that manages this FlutterViewController.
+ *
+ * A FlutterViewController can be initialized either with an already-running `FlutterEngine`,
+ * or it can be initialized with a `FlutterDartProject` that will be used to spin up
+ * a new `FlutterEngine`.  Developers looking to present and hide FlutterViewControllers
+ * in native iOS applications will usually want to maintain the `FlutterEngine` instance
+ * so as not to lose state or asynchronous tasks when navigating back and forth between a
```

Comment:
specify the dart-side nature of 'state' and 'task'. i.e. if you subclass FlutterViewController and put stuff there in the objective-c class instance, that's still lost.